elastic/beats
# [Heartbeat] Fix reload policy hash consistency, browser monitor cleanup and semaphore release

## What does this PR do?

Closes #34629.

Addresses multiple issues with private location updates:
- Remove global fields from policy to prevent agent from stopping integrations which have not been updated.
- Detect browser monitor has ben cancelled by having a global (to the monitor) context .
- Fix an issue where job limit semaphore was not released after being acquired.

## Checklist


## How to test this PR locally



1. Create a private location policy with multiple long-running browser monitors.
```js
step("Test", async () => {
    await page.goto("https://google.es");
    await page.waitForTimeout(600e3);
})
```
2. Enroll an agent in the policy.
3. On a different window, monitor processes executed by agent-heartbeat. This is an example to monitor a docker instance:
```zsh
$: docker exec -it -u root agent watch ps -aef
```
4.  Start updating monitors in Kibana (can be by enabling/disabling them) one by one and watch for agent reaction.
    - When a monitor that is running is disabled, heartbeat should kill the synthetics process and schedule the next job.
    - When a monitor that is **not** running is updated, heartbeat should not terminate any of the running jobs.
    - After multiple updates, heartbeat should still continue scheduling jobs since the job limit semaphore is updated even when a job is cancelled.
